### PR TITLE
[one-cmds] Revise preprocess_images.py to process also numpy files

### PR DESCRIPTION
**Pull request by Seunghui98**
This commit revises preprocess_images.py to process also numpy files.
- preprocess_images_numpy.py is very similar to preprocess_images.py. so, merge them to reduce duplicate codes.

ONE-DCO-1.0-Signed-off-by: SeungHui Lee <shsh1004.lee@samsung.com>

---
Related Issue: https://github.com/Samsung/ONE/issues/12053
Draft PR: https://github.com/Samsung/ONE/pull/12186

### CODE REVIEW

File: compiler/one-cmds/tests/prepare_test_materials.sh [hunk 1]
```diff
@@ -57,7 +57,7 @@ if [[ ! -s "img_files" ]]; then
     # https://github.com/Samsung/ONE/issues/3213#issuecomment-722757499
 fi
 
-if [ ! -d "raw_files" ] || [ ! -s "datalist.txt" ]; then
+if [ ! -d "raw_files" ] || [ ! -s "datalist.txt" ] || [ ! -d "numpy_files" ] || [ ! -s "datalist_numpy.txt" ]; then
```
Comment (seanshpark) [inline compiler/one-cmds/tests/prepare_test_materials.sh:60]: @jinevening , can we use `datalist.txt` for both raw and numpy files?

File: compiler/one-cmds/tests/prepare_test_materials.sh [hunk 2]
```diff
@@ -68,7 +68,7 @@ if [[ ! -s "inception_v3_test_data.h5" ]]; then
 fi
 
 if [[ ! -d "test_saved_model" ]]; then
-    rm -rf test_saved_model.zip
+    rm -rf test_saved_model.zipo
```
Comment (seanshpark) [inline compiler/one-cmds/tests/prepare_test_materials.sh:71]: ?

File: compiler/one-cmds/tests/prepare_test_materials.sh (hunk 1 above)
Comment (jinevening) [reply compiler/one-cmds/tests/prepare_test_materials.sh:60]: I think we can't. The names of data files are different for raw and numpy files, so we will need different list files.

Comment (seanshpark) [reply compiler/one-cmds/tests/prepare_test_materials.sh:60]: @Seunghui98 , how is this change related with two python scripts?

File: compiler/one-cmds/tests/prepare_test_materials.sh (hunk 2 above)
Comment (Seunghui98) [reply compiler/one-cmds/tests/prepare_test_materials.sh:71]: Oh.. I am sorry that i made a mistake. 😅
Thank you a lot!

File: compiler/one-cmds/tests/prepare_test_materials.sh (hunk 1 above)
Comment (Seunghui98) [reply compiler/one-cmds/tests/prepare_test_materials.sh:60]: > how is this change related with merge of two python scripts?

I think you may have felt that the intermediate process for this change is missing.

In fact, the following code should have been included in CMakeLists.txt, in this PR. https://github.com/Samsung/ONE/pull/12283
but by my mistake, I forgot to put the code and uploaded pr, but it was merged. 😅
```
if [ ! -d "numpy_files" ] || [ ! -s "datalist_numpy.txt" ]; then
    ../bin/venv/bin/python preprocess_images_numpy.py
fi
```

Maybe if the code went in without any mistakes, **this change** would have been as follows.
```
if [ ! -d "raw_files" ] || [ ! -s "datalist.txt" ]; then
    ../bin/venv/bin/python preprocess_images.py
fi

if [ ! -d "numpy_files" ] || [ ! -s "datalist_numpy.txt" ]; then
    ../bin/venv/bin/python preprocess_images_numpy.py
fi

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> merge two scripts

if [ ! -d "raw_files" ] || [ ! -s "datalist.txt" ] || [ ! -d "numpy_files" ] || [ ! -s "datalist_numpy.txt" ]; then
    ../bin/venv/bin/python preprocess_images.py
fi
```

Comment (Seunghui98) [reply compiler/one-cmds/tests/prepare_test_materials.sh:60]: I think this change in `CMakelist.txt` is not related to PR content.
so, I will remove this from PR now and upload it in the next PR. :)

Comment (seanshpark) [reply compiler/one-cmds/tests/prepare_test_materials.sh]: I think we should land the fix before this PR.

Comment (Seunghui98) [reply compiler/one-cmds/tests/prepare_test_materials.sh]: @seanshpark 

Okay. i will upload PR.
Thank you a lot :)

**Comment by jinevening** [comment]
#12352 is related to this PR.

**Comment by Seunghui98** [comment]
@seanshpark @jinevening @mhs4670go 

The previous PR (https://github.com/Samsung/ONE/pull/12352) has become merge. 
so, Could you review the code again? :)

File: compiler/one-cmds/tests/preprocess_images.py [hunk 3]
```diff
@@ -32,7 +39,16 @@
 
             img = np.array(image.resize((299, 299), ANTIALIAS)).astype(np.float32)
             img = ((img / 255) - 0.5) * 2.0
-            output_file = output_dir + '/' + f.replace('jpg', 'data')
-            img.tofile(output_file)
-            datalist.writelines(os.path.abspath(output_file) + '\n')
-    datalist.close()
+
```
Comment (seanshpark) [inline compiler/one-cmds/tests/preprocess_images.py:42]: plz split this adding new line to another PR

**Comment by seanshpark** [comment]
There are two changes, 1) revise `preprocess_images.py` to process also for numpy files 2) deprecate `preprocess_images_numpy.py` file. Plz split these to separate PRs

**Comment by seanshpark** [comment]
plz try to `make install` with this change

**Comment by Seunghui98** [comment]
@seanshpark 

> plz try to make install with this change

I didn't understand what to do about `make install`. 😅
By guess, am I doing the following..?
```
$  NNAS_BUILD_PREFIX=build/release ./nnas create-package --preset 20230907 --prefix build/install
```

**Comment by seanshpark** [comment]
> I didn't understand what to do about make install

`./nncc build install`

follow `nncc` and `build`. read documents about `cmake` to install a project.

**Comment by Seunghui98** [comment]
@seanshpark 

Thank you a lot :)
I tried to do `./nncc build install` to test this change.
As a result, The test of `one-create-quant-dataset` were all successful.

**Comment by seanshpark** [comment]
close/reopen to trigger tests...

**Review by seanshpark — APPROVED** [review]
LGTM

**Comment by seanshpark** [comment]
Q) `../bin/venv/bin/python preprocess_images_numpy.py` is still called. is this OK?

**Comment by Seunghui98** [comment]
@seanshpark 

> Q) ../bin/venv/bin/python preprocess_images_numpy.py is still called. is this OK?

yes. :)
When `preprocess_images.py` is executed, `numpy_files` and `datalist_numpy.txt` files are created.
Then check if there are `numpy_files` and `datalist_numpy.txt` before `preprocess_images_numpy.py` is run and if not, 
`preprocess_images_numpy.py` is run.

Next time, I will upload the PR that removes`preprocess_images_numpy.py` from `preprocess_images_numpy.py`.

**Comment by Seunghui98** [comment]
@jinevening @mhs4670go 

Could you review this PR...? :)

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by jinevening — APPROVED** [review]
LGTM
